app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ArticlesController < ApplicationController
+  before_action :find_article, except: %i[index new create]
+
+  http_basic_authenticate_with name: 'user', password: 'secret', except: %i[index show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article 
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+    begin
+      @article.save!
+    rescue StandardError => e
+      render :new
+    else
+      redirect_to(article_path(@article))
+    end
+  end
+
+  def edit
+    @article
```

Comment:
Is it necessary to return the @article variable?

Reply:
[Same answer](https://github.com/SanGeeky/rails-training/pull/1#discussion_r611911896)

Should I delete the `edit` method? Because at this moment the method only returns the article.

Reply:
See https://github.com/SanGeeky/rails-training/pull/1#discussion_r612080487

Reply:
Thanks!